dms/models/res_users.py

```diff
@@ -0,0 +1,151 @@
+            )
+            dir_ids |= set(children_directories.ids)
+        return dir_ids
+
+    def _get_directories_from_access_group(self, operation):
+        """
+        Get allowed directory ids for the user based on its inclusion in an access
+        group.
+
+        :param str operation: operation to check
+        :return: Allowed directory ids
+        :rtype: set[int]
+        """
+        domain = [
+            ("users", "in", self.env.uid),
+        ]
+        if operation != "perm_read":
+            domain += [(operation, "=", True)]
+        directories_from_access_group = (
+            self.env["dms.access.group"]
+            .search(domain)
+            .directory_ids.filtered(
+                lambda x: not x.is_hidden
+                and not x.storage_id_inherit_access_from_parent_record
+            )
+        )
+        return set(directories_from_access_group.ids)
+
+    @api.model
+    def _has_all_access(self):
+        """
+        Check if user has access to all directories for the specified operation.
+
+        :return: The access group that has all access
+        :rtype: odoo.model.dms_access_group
+        """
+        if self.has_group("dms.group_dms_manager"):
+            return self.env["dms.access.group"].sudo().search([])
+        return self.env["dms.access.group"]
+
+    def _get_directory_from_attachment(self, operation):
+        """
+        Get the directory ids from the attachments that the user has access to. The
+        goal here is to allow the user to
+        access the directories based on the rule given in its profile settings. That
+        way, we are only using the standard
+        rule checking.
+
+        :param str operation: operation to check
+
+        :return: The directory ids that the user has access to
+        :rtype: list[int]
+        """
+
+        # get the directories based on a record
+        attachment_directories = self.env["dms.directory"].search(
+            [
+                ("storage_id_inherit_access_from_parent_record", "=", True),
+            ]
+        )
+        if not attachment_directories:
+            return []
+
+        authorized_directory_ids = []
+        for directory in attachment_directories:
+            # use of .get because on the first startup of Odoo, all the modules might
+            # not be loaded yet
+            model = self.env.get(directory.res_model)
+            if model is not None:
+                record = model.browse(directory.res_id).exists()
+                authorized = True
+                try:
+                    record.with_env(
+                        self.env(su=self.env.user._is_superuser())
+                    ).check_access_rule(operation.replace("perm_", ""))
+                except AccessError:
+                    authorized = False
+
+                if authorized:
+                    authorized_directory_ids.append(directory.id)
+
+        return authorized_directory_ids
```

Comment:
IMO I still think this is not necessary (in any case it would be an improvement, not something necessary for migration). This change would be replicating what already exists in the defined `ir.rules`.

Reply:
This is still pending

Reply:
Ok, you forgot to remove all the `res.users` code which is now no longer needed.

Reply:
Yeah, you're right, my bad. I removed it, but when I pushed I had a conflit. 
I removed it (again).